pilot/pkg/networking/plugin/mixer/mixer.go

```diff
@@ -246,9 +236,14 @@ func buildHTTPMixerFilterConfig(mesh *meshconfig.MeshConfig, role model.Proxy, n
 		addStandardNodeAttributes(mxConfig.ForwardAttributes.Attributes, v1.AttrSourcePrefix, role.IPAddress, role.ID, labels)
 	}
 
+	// gateway case is special because upstream listeners are considered outbound, however we don't want to
+	// automatically disable policy / report.
+	disablePolicy := (outboundRoute && role.Type != model.Router) || mesh.DisablePolicyChecks
+	disableReport := outboundRoute && role.Type != model.Router
```

Comment:
A nit, I would change it to:
```
disableReport := outboundRoute && role.Type != model.Router
disablePolicy := disableReport || mesh.DisablePolicyChecks
```

Reply:
Even though that eliminates the code duplication it implies a relationship between the two vars which doesn't exist. For that reason I'd prefer to live with the duplication.

Reply:
Ah, I see, there is no logical relationship. A possible way to reduce the code duplication, maybe an overkill:
```go
nonRouterOutboundRoute := outboundRoute && role.Type != model.Router
disablePolicy := nonRouterOutboundRoute || mesh.DisablePolicyChecks
disableReport := nonRouterOutboundRoute 
```